proposals/NNNN-unicode-scalar-properties.md

```diff
@@ -0,0 +1,370 @@
+
+On Linux, system ICU libraries are built with function renaming enabled (the
+default), so function names have the `_NN` version number suffix. Function
+renaming makes it more difficult to use these APIs from Swift; even though the C
+header files contain `#define`s that map function names like `u_foo_59` to
+`u_foo`, these `#define`s are not imported into Swift—only the suffixed function
+names are available. This means that Swift bindings would be fixed to a specific
+version of the library without some other intermediary layer. Again, this is
+significant overhead.
+
+Therefore, this proposal not only fills in important gaps in the standard
+library's capabilities, but removes a significant pain point for users who may
+try to access that functionality through other means.
+
+## Proposed solution
+
+We propose adding a nested struct, `Unicode.Scalar.Properties`, which will
+encapsulate many of the properties that the Unicode specification defines on
+scalars. Supporting types, such as enums representing the values of certain
+properties, will also be added to the `Unicode` enum "namespace."
+
+### Scope of This Proposal
+
+This proposal is restricted, by design, to add functionality to `Unicode.Scalar`
+only. While we believe that some of the properties described here (and others)
+would be valuable on `Character` as well, we have intentionally saved those for
+a future proposal in order to keep this one small and focused. Such a future
+proposal would likely depend on the design and implementation herein.
+
+## Detailed design
+
+The code snippets below reflect an elided sketch of the proposed public API
+only. Full details can be found in the implementation pull request.
+
+In general, the names of the properties inside the `Properties` struct are
+derived directly from the names of the properties as they are defined in the
+[Unicode Standard](http://unicode.org/reports/tr44/#Property_Index).
+
+```swift
+extension Unicode.Scalar {
+
+  // NOT @_fixed_layout
+  public struct Properties {
+    // Remaining API is defined in the subsections below.
+  }
+
+  /// The value that encapsulates the properties exposed 
+  public var properties: Properties { get }
+}
+```
+
+### Boolean Properties
+
+Each of the Boolean properties in the first block below would be implemented by
+calling `u_hasBinaryProperty` with the property name indicated to the right of
+the computed property.
+
+We propose supporting all of the Boolean properties that are currently available
+using ICU's `u_hasBinaryProperty` that correspond to properties in the Unicode
+Standard, but _not_ ICU-specific properties, with the following exceptions:
+
+* `UCHAR_GRAPHEME_LINK` is omitted because it is deprecated and equivalent to
+  canonical combining class 9.
+* `UCHAR_HYPHEN` is omitted because is deprecated in favor of the `Line_Break`
+  property.
+
+```swift
+extension Unicode.Scalar.Properties {
+
+  public var isAlphabetic: Bool { get }    // UCHAR_ALPHABETIC
```

Comment:
For this proposal, probably better to list the UCD name rather than the ICU API. (I believe the implementation PR has already been adjusted).

Reply:
Done.